### PR TITLE
Fix multilevel IBFE when the structure is not on the finest level.

### LINKED ISSUE #1112: Multilevel IBFE support
We talked about this a bit on slack and it's worth summarizing the status quo here.

The main goal is to let different `Elem`s interact with different levels in the patch hierarchy. The main application of this at the moment is putting valves in the heart model on some extra-special level while big structures are on a coarser level.

Users should be able to enable this feature by adding a new section to their input file:
```
IBFEMethod {
   // ... normal stuff goes here

   // this part is new
   FEDataManager {
      block_ids_on_levels {
         level_0 = // nothing on level 0
         level_1 = 100, 110 // big structures that aren't that dynamic are in the middle
         level_2 = // everything else on level 2
      }
   }
}
```
the idea is that we can keep backwards compatibility by assigning everything to the finest level by default. Unlike sidesets, block ids are unique to an element.

Here is our progress so far:
- [x] Give FEDataManager an input database (#1089)
- [x] Cleanup `FEDataManager` so that we do not need to worry about almost half a dozen constructors (#1093, #1104)
- [x] Modify `FEDataManager`'s data structures so that we loop over all levels instead of just the finest level (#1108)
- [x] Implement the `block_ids_on_levels` (or `subdomain_ids_on_levels`: the terminology isn't that important) input database and set up an appropriate data structure inside `FEDataManager` to store it (#1113)
- [x] Implement a function in FEDataManager that looks up what level an element should be on from its subdomain id (#1113)
- [x] Reimplement cell tagging based on what level an element should be on (#1119).
- [x] Reimplement the way we associate patches with elements (#1123).

At this point velocity interpolation should work automatically.
- [x] Correctly prolong quadrature point data so that levels are tagged correctly (#1127)
- [x] Implement some method for prolonging forces spread onto coarser patch levels up to finer patch levels as appropriate. This is the only step I don't know how to do yet (#1129)
- [x] Make sure things work when the mesh is *not* on the finest mesh level. (#1140)
- [x] Update the scratch hierarchy in IBFEMethod to work with multiple levels (this might be hard) (#1129, #1136)

### PR DESCRIPTION
**Pull request by drwells**
Closes #1112.

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

File: ibtk/src/lagrangian/FEDataManager.cpp [hunk 1]
```diff
@@ -3025,6 +3050,8 @@ FEDataManager::collectActivePatchElements(std::vector<std::vector<Elem*> >& acti
     const MeshBase& mesh = d_fe_data->d_es->get_mesh();
     System& X_system = d_fe_data->d_es->get_system(COORDINATES_SYSTEM_NAME);
 
+    // We may
```
Comment (boyceg) [inline ibtk/src/lagrangian/FEDataManager.cpp:3053]: ??

Comment (drwells) [reply ibtk/src/lagrangian/FEDataManager.cpp:3053]: I was going to write a comment here but it ended up not being necessary. Let me delete the stub.

**Review by boyceg — APPROVED** [review]
Looks good.

Should we check somewhere that the finest patch hierarchy level is >= the max level number?

**Comment by drwells** [comment]
I don't think so. In the case of the test I added we may not necessarily even have patches on the finest patch level until later in the time evolution since the structure isn't on it.

**Comment by boyceg** [comment]
You can use REFINE_BOXES to make sure there are finer levels.

**Comment by drwells** [comment]
Good idea. I'll add another test that does exactly that.

**Pull request merged**